### PR TITLE
feat(vital-card): JEPZ-406 | Category Card 2.0

### PR DESCRIPTION
**Pull request by samlynndavis**

### CODE REVIEW

File: packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import {
+  DefaultDomains, vitalCategoryCardElement, vitalSpacing, vitalTypography
+} from '@bodiless/vital-elements';
+import { vitalLink } from '@bodiless/vital-link';
+import { CardComponents } from 'src/components/Card';
+import { CardToken } from 'src/components/Card/CardClean';
+import Base from '../../Card/tokens/Base';
+import { asCategoryCardToken } from '../CategoryCardClean';
+
+// const Default = asCategoryCardToken({
```
Comment (hvanyo) [inline packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts:15]: @samlynndavis  lets remove comments if you are not using them

File: packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+const Default = asCategoryCardToken({
+  ...Base,
+  Components: {
+    ...Base.Components,
+    Eyebrow: undefined,
+    Description: undefined,
+    CTAWrapper: undefined,
+    CTALink: vitalLink.Default,
+  },
+  Theme: {
```
Comment (hvanyo) [inline packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts:96]: @samlynndavis  is your intent to override Theme, Layout, Spacing, Content for CategoryCard?

File: packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    ContentWrapper: 'items-start',
+  },
+  Layout: {
+    ImageWrapper: 'relative pb-[60%] overflow-hidden',
```
Comment (hvanyo) [inline packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts:100]: @samlynndavis  
why is pb- in Layout and not spacing token?  
why is imagewrapper/image relative/absolute?  It seems unnecessary

File: packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+  }
+});
+
+export interface VitalCategoryCard extends TokenCollection<CardComponents, DefaultDomains> {
+  /**
+   * Defines the default Article card for the Vital DS.
+   * - Extends the Base card.
+   */
+  Default: CardToken,
+  /**
```
Comment (hvanyo) [inline packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts:143]: @samlynndavis  can you delete the API doc you are not using and update Default & WithTitle

File: packages/vital-card/src/shadow/@bodiless/vital-card/CategoryCard.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+// import { shadow } from '@bodiless/vital-elements';
```
Comment (hvanyo) [inline packages/vital-card/src/shadow/@bodiless/vital-card/CategoryCard.ts:1]: why this file if its all comments/?

File: packages/vital-demo/src/styleguide/Page/StyleGuideTemplate/CategoryCard.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+import {
+  flowHoc, replaceWith, on, varyDesigns, addProps
+} from '@bodiless/fclasses';
+
+import { StyleGuideExamplesClean, vitalStyleGuideExamples } from '../../Examples';
+
+/*
+ * Compose the default vertical & horizontal cards
+ */
+const CategoryCardVariations = varyDesigns(
```
Comment (hvanyo) [inline packages/vital-demo/src/styleguide/Page/StyleGuideTemplate/CategoryCard.tsx:14]: @samlynndavis  you didn't vary on anything CategoryCardVariations or CategoryCardVariations you just composed to components.

`
const Default = on(CategoryCardClean)(vitalCategoryCard.Default);
const TitleVariations = varyDesigns(
  {
    '': '', // vary on itself so you get the Default variation
    WithTitle: vitalCategoryCard.WithTitle,
  },
  Default,
);
`

File: packages/vital-link/src/components/Link/tokens/vitalLink.ts [hunk 7]
```diff
@@ -23,7 +23,7 @@ import {
   flowHoc,
 } from '@bodiless/fclasses';
 import { withSidecarNodes, withNodeKey } from '@bodiless/data';
-import { vitalColor, vitalTypography } from '@bodiless/vital-elements';
+import { vitalColor, vitalLinkElement } from '@bodiless/vital-elements';
```
Comment (hvanyo) [inline packages/vital-link/src/components/Link/tokens/vitalLink.ts:26]: @samlynndavis  why are these changes here in category card/?

File: packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts (hunk 3 above)
Comment (samlynndavis) [reply packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts:100]: The design requires the image to maintain a 5:3 aspect ratio. I was initially using the padding percentage trick to achieve this (now using tw for that), but need to position the image absolutely to keep it within the bounds of the wrapper.

File: packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts (hunk 4 above)
Comment (samlynndavis) [reply packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts:143]: Done.

File: packages/vital-card/src/shadow/@bodiless/vital-card/CategoryCard.ts (hunk 5 above)
Comment (samlynndavis) [reply packages/vital-card/src/shadow/@bodiless/vital-card/CategoryCard.ts:1]: This is part of a bit of cleanup that I hadn't yet finished. I'll address this.

File: packages/vital-link/src/components/Link/tokens/vitalLink.ts (hunk 7 above)
Comment (samlynndavis) [reply packages/vital-link/src/components/Link/tokens/vitalLink.ts:26]: I presumed that the link was to be used on this card and was adding to see how it would look. After our discussion, I'll be switching to the button. This will be removed.

File: packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts (hunk 3 above)
Comment (hvanyo) [reply packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts]: @samlynndavis  where do designers specify the 5:3 aspect ratio?   We didn't talk about this requirement with design.  The image should just be w-full & thus will take be whatever height it is uploaded.

File: packages/vital-demo/src/styleguide/Page/StyleGuideTemplate/CategoryCard.tsx [hunk 8]
```diff
@@ -9,26 +9,24 @@ import {
...
  */
-const CategoryCardVariations = varyDesigns(
-  {
-    Default: on(CategoryCardClean)(vitalCategoryCard.Default),
-  },
-);
+const Default = {
+  Default: on(
+    CategoryCardClean
+  )(vitalCategoryCard.Default), // vary on itself so you get the Default variation
```
Comment (hvanyo) [inline packages/vital-demo/src/styleguide/Page/StyleGuideTemplate/CategoryCard.tsx:17]: I don't think you need comment here it was intended for the Variations.  @samlynndavis

File: packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    Title: undefined,
+    CTALink: on(ButtonClean)(CategoryCardButton),
+  },
+  Layout: {
+    ImageWrapper: 'relative overflow-hidden aspect-5/3',
+    Image: 'absolute top-0 h-full w-full',
+    ContentWrapper: 'items-start',
+  },
+  Spacing: {
+    ContentWrapper: 'py-16px md:py-24px',
```
[the remark this answers is not in the document]
Comment (samlynndavis) [reply packages/vital-card/src/components/CategoryCard/tokens/vitalCategoryCard.ts:44]: I completely forgot that those tokens were coming in the new iteration of the script. Thank you for reminding me. I'll place that in the token collection and use that now.

**Review by hvanyo — APPROVED** [review]

**Review by hvanyo — APPROVED** [review]

**Review by VancheeZze — APPROVED** [review]

**Review by VancheeZze — DISMISSED** [review]

**Review by KaterinaPashkevich — DISMISSED** [review]

**Review by KaterinaPashkevich — APPROVED** [review]
